### PR TITLE
Add option to show active points on line and multi-lines charts

### PR DESCRIPTION
**Pull request by thomaschampagne**
This PR includes:

- Option to show active points on line and multi-lines charts
- Documentation samples to use this new feature.

**Screenshot of the added documentation (tab "data"):**
image

Hope this helps!

Thomas

### CODE REVIEW

**Comment by wlach** [comment]
Thanks for the patch! I will try to make some time to look at it tomorrow.

File: src/js/charts/line.js [hunk 1]
```diff
@@ -544,6 +549,25 @@
 
       args.data[i].line_id = line_id;
 
+      // If option activated, add active points for each lines
+      if (args.active_point_on_lines) {
+
+        svg.selectAll('circle-' + line_id)
+          .data(args.data[i])
```
Comment (hamilton) [inline src/js/charts/line.js]: Is there a way we can actually subset `args.data[i]` to produce only the data points we need? This d3 call will produce extra circles for every data point even if it is not visible, which will add a bunch of weight to the DOM that is probably not needed. We are already committing this sin in MG, so I'm hoping to keep us from bringing in an even heavier DOM down the line.

Comment (thomaschampagne) [reply src/js/charts/line.js]: You right. It's a bit dirty :/. I will try to fix that this weekend.
The use case is the same than my 2 other PRs => https://twitter.com/champagnethomas/status/972161893384228866

Comment (hamilton) [reply src/js/charts/line.js]: Great - thanks @thomaschampagne!

Comment (hamilton) [reply src/js/charts/line.js]: btw, really neat to see how you're using MG. Don't hesitate to send along more screenshots.

Comment (thomaschampagne) [reply src/js/charts/line.js]: @hamilton I can suggest that improvement:

```javascript
      if (args.active_point_on_lines) {
        svg.selectAll('circle-' + line_id)
          .data(args.data[i])
          .enter()
          .filter((d) => {
            return d.active;
          })
          .append('circle')
          .attr('class', 'mg-area' + (line_id) + '-color mg-shown-active-point')
          .attr('cx', args.scalefns.xf)
          .attr('cy', args.scalefns.yf)
          .attr('r', () => {
            return args.active_point_size;
          });
      }
```
I introduced a `filter()` between `enter()` and `append('circle')`. With this code only active points will create 'circle' elements in the DOM . Tested and verified on my side.

Let me know if this solution is sufficient for you.

**Comment by thomaschampagne** [comment]
@hamilton @wlach Does my latests changes (https://github.com/mozilla/metrics-graphics/pull/827/commits/82bc9179da724b0d5e8b8a7dc5263b930369d2ed) are OK for you? Or i have to review my work?

**Comment by wlach** [comment]
Sorry for the delay on reviewing this. Everything looks good to me here, I will merge!

**Pull request merged**